starfish/imagestack/imagestack.py

```diff
@@ -399,19 +384,9 @@ def sel(self, indexers: Mapping[Axes, Union[int, tuple]]):
         """
 
         # convert indexers to Dict[str, (int/slice)] format
+        stack = deepcopy(self)
```

Comment:
Whoa -- this is going to make a full copy of the ImageStack to extract a subset. Are we sure this is necessary?

Reply:
yea you don't want to modify the existing imagestack. The old method created a brand new imagestack so I don't this is too crazy of a difference

Reply:
Tried to change this to just a basic copy(self) but ran into an internal pandas error. I think after tony's changes to the image stack class land though we should be able to use xarray.copy(deep=false) and save some memory.

Reply:
```suggestion
        # TODO shanaxel42 check if this can be changed to xarray.copy(deep=false)
        stack = deepcopy(self)

Reply:
What sort of pandas error?